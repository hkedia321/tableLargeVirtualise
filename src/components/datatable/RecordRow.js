import React from "react"
import PropTypes from "prop-types"
import DataCellGraph from "./DataCellGraph"
import DataCellCampaignId from "./DataCellCampaignId"

const RecordRow = React.memo((props) => {
  const {
    rowData,
    dataTableGraphRecords,
    isCampaignColumnOpen,
    isGraphColumnsOpenArray,
  } = props
  return (
    <tr className="rowDiv" data-testid="recordRow" style={props.style}>
      <DataCellCampaignId
        text={rowData.groups.Campaign.metadata.name}
        isOpen={isCampaignColumnOpen}
      />
      {dataTableGraphRecords.map((columnObject, index) => (
        <DataCellGraph
          key={index}
          isOpen={isGraphColumnsOpenArray[index]}
          columnObject={columnObject}
          rowData={rowData}
        />
      ))}
    </tr>
  )
})

RecordRow.propTypes = {
  dataTableGraphRecords: PropTypes.array.isRequired,
  rowData: PropTypes.object.isRequired,
  isCampaignColumnOpen: PropTypes.bool.isRequired,
  isGraphColumnsOpenArray: PropTypes.array.isRequired,
  style: PropTypes.object.isRequired,
}

export default RecordRow
