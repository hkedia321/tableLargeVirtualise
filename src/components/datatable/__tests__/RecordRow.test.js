import React from "react"
import { render, cleanup } from "@testing-library/react"
import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import "jest-styled-components"
import RecordRow from "../RecordRow"

jest.mock("react-tooltip", () => () => <div>react-tooltip</div>)

jest.mock("highcharts-react-official", () => {
  // eslint-disable-next-line react/prop-types
  return function DummyHighcharts({ options }) {
    return <div id="MockedHighcharts">{JSON.stringify(options)}</div>
  }
})

afterEach(cleanup)

const recordRowProps = {
  isCampaignColumnOpen: true,
  isGraphColumnsOpenArray: [true, true],
  rowData: {
    groups: {
      Campaign: {
        metadata: {
          name: "713a8eee-3922-403a-9783-0db899c54ead",
        },
      },
    },
    trend: [
      {
        day: "2020-06-01",
        aj_coh_0w_ios_real_acquisition: 0,
        aj_app_and_installs: 2496,
        aj_app_ios_installs: 0,
        aj_coh_0w_and_real_acquisition: 182,
      },
      {
        day: "2020-06-02",
        aj_coh_0w_ios_real_acquisition: 0,
        aj_app_and_installs: 2476,
        aj_app_ios_installs: 0,
        aj_coh_0w_and_real_acquisition: 167,
      },
    ],
  },
  dataTableGraphRecords: [
    {
      heading: "App and Installs",
      key: "aj_app_and_installs",
    },
    {
      heading: "CoH Real Acquisition",
      key: "aj_coh_0w_and_real_acquisition",
    },
  ],
}

it("renders recordRow correctly with showing all columns", () => {
  const tbody = document.createElement("tbody")
  const { getByTestId, queryByTestId, queryAllByTestId } = render(
    <RecordRow {...recordRowProps} />,
    {
      container: document.body.appendChild(tbody),
    }
  )
  expect(getByTestId("recordRow"))
  // campaign cell should show the data
  expect(queryByTestId("dataCellCampaignId")).toHaveTextContent(
    recordRowProps.rowData.groups.Campaign.metadata.name
  )
  // graph cells should show the data
  // eslint-disable-next-line no-restricted-syntax
  for (const [index] of recordRowProps.dataTableGraphRecords.entries()) {
    expect(queryAllByTestId("dataCellGraph")[index].firstChild)
  }
})

it("renders recordRow correctly when campaign id cells are hidden", () => {
  const tbody = document.createElement("tbody")
  const { queryByTestId, queryAllByTestId } = render(
    <RecordRow {...recordRowProps} isCampaignColumnOpen={false} />,
    {
      container: document.body.appendChild(tbody),
    }
  )
  // campaign cell should not show the data
  expect(queryByTestId("dataCellCampaignId")).not.toHaveTextContent(
    recordRowProps.rowData.groups.Campaign.metadata.name
  )
  // graph cells should show the data
  // eslint-disable-next-line no-restricted-syntax
  for (const [index] of recordRowProps.dataTableGraphRecords.entries()) {
    expect(queryAllByTestId("dataCellGraph")[index].firstChild)
  }
})

it("renders recordRow correctly when graph cells are hidden", () => {
  const tbody = document.createElement("tbody")
  const { queryByTestId, queryAllByTestId } = render(
    <RecordRow
      {...recordRowProps}
      isGraphColumnsOpenArray={new Array(
        recordRowProps.dataTableGraphRecords.length
      ).fill(false)}
    />,
    {
      container: document.body.appendChild(tbody),
    }
  )
  // campaign cell should show the data
  expect(queryByTestId("dataCellCampaignId")).toHaveTextContent(
    recordRowProps.rowData.groups.Campaign.metadata.name
  )
  // graph cells should not show the data
  // eslint-disable-next-line no-restricted-syntax
  for (const [index] of recordRowProps.dataTableGraphRecords.entries()) {
    expect(queryAllByTestId("dataCellGraph")[index].firstChild).toBeNull()
  }
})

it("matches snapshot", () => {
  const tbody = document.createElement("tbody")
  const tree = renderer
    .create(<RecordRow {...recordRowProps} />, {
      container: document.body.appendChild(tbody),
    })
    .toJSON()
  expect(tree).toMatchSnapshot()
})
