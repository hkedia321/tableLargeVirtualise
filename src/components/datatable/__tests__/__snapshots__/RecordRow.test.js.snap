// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c4 {
  position: absolute;
  z-index: 1;
  font-weight: normal;
  right: 20px;
  top: -5px;
  text-align: right;
  color: #3f3f3f;
}

.c3 {
  position: relative;
  padding: 10px;
  width: auto;
  padding-top: 30px;
}

.c0 {
  position: relative;
  padding: 20px;
}

.c1 {
  padding-right: 10px;
  width: 20px;
}

.c2 {
  width: calc(100% - 25px);
  display: inline-block;
}

<tr
  className="rowDiv"
  data-testid="recordRow"
>
  <div
    className="c0 dataCellCampaignId"
    data-testid="dataCellCampaignId"
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-angle-right fa-w-8 fa-lg c1"
      color="#3F51B5"
      data-icon="angle-right"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 256 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <span
      className="c2"
    >
      713a8eee-3922-403a-9783-0db899c54ead
    </span>
  </div>
  <div
    className="c3 dataCellGraph"
    data-testid="dataCellGraph"
  >
    <h4
      className="c4"
      dangerouslySetInnerHTML={
        Object {
          "__html": "&euro;&nbsp;4,972",
        }
      }
    />
    <div
      id="MockedHighcharts"
    >
      {"chart":{"type":"area","height":"43.75%","margin":0},"credits":{"enabled":false},"title":false,"legend":{"enabled":false},"xAxis":{"labels":{"enabled":false},"title":false,"tickLength":0,"lineColor":"transparent"},"yAxis":{"labels":{"enabled":false},"title":false,"gridLineWidth":0},"plotOptions":{"series":{"fillColor":"#eaebfe","color":"#c8c9ff","marker":{"radius":2.5}}},"tooltip":{},"series":[{"data":[["2020-06-01",2496],["2020-06-02",2476]]}]}
    </div>
  </div>
  <div
    className="c3 dataCellGraph"
    data-testid="dataCellGraph"
  >
    <h4
      className="c4"
      dangerouslySetInnerHTML={
        Object {
          "__html": "&euro;&nbsp;349",
        }
      }
    />
    <div
      id="MockedHighcharts"
    >
      {"chart":{"type":"area","height":"43.75%","margin":0},"credits":{"enabled":false},"title":false,"legend":{"enabled":false},"xAxis":{"labels":{"enabled":false},"title":false,"tickLength":0,"lineColor":"transparent"},"yAxis":{"labels":{"enabled":false},"title":false,"gridLineWidth":0},"plotOptions":{"series":{"fillColor":"#eaebfe","color":"#c8c9ff","marker":{"radius":2.5}}},"tooltip":{},"series":[{"data":[["2020-06-01",182],["2020-06-02",167]]}]}
    </div>
  </div>
</tr>
`;
